src/odemis/acq/stream/_helper.py

```diff
@@ -168,13 +170,38 @@ def __init__(self, name, detector, dataflow, emitter, scanner=None, sstage=None,
     # Overrides method of LiveStream
     def _onNewData(self, dataflow, data):
         """
-        Called when a new image has arrived from the detector.
-        Usually the dataflow subscribes to onNewData.
+        Called when a new image has arrived from the detector. Usually the dataflow subscribes to onNewData.
+        If there is integrationTime and there are more than 1 image to be summed, integrate one image after another
+        for live display while running a stream.
         :param dataflow: (model.Dataflow) The dataflow.
         :param data: (model.DataArray). The new image that has arrived from the detector.
         """
-        if hasattr(self, "integrationTime"):
-            self.raw = ([data] + self.raw)[:self.integrationCounts.value]
+        if hasattr(self, "integrationTime") and self.integrationCounts.value > 1:
+            self._integration_step += 1
+            # Once the value of integration counts is reached, prepare the raw data to store the
+            # first acquired image for the next ebeam position. Handle also the fact that integrationCounts
+            # can change while playing by resetting everything.
+            if (self._img_intor and self._img_intor.steps != self.integrationCounts.value) or\
+                    self._integration_step > self.integrationCounts.value:
+                self._img_intor = None
+                self.raw = [data]
+                self._integration_step = 1
+            else:
+                self.raw = (self.raw + [data])
+
+            # TODO: when increasing the integration count(compared to before), keep the previous accumulated images.
+            # integrate the images one after another for live display
+            if self._img_intor is None:
+                self._img_intor = img.ImageIntegrator(self.integrationCounts.value)
+            # Catch the exception when the user changes some of the hardware settings (eg, resolution), while playing
+            # the stream. To do so, restart everything and keep the last image received.
+            try:
+                self.raw = self._img_intor.append(self.raw)
+            except Exception:
+                logging.warning("Failure when updating the hardware settings")
+                self._img_intor = None
+                self.raw = [data]
+                self._integration_step = 1
```

Comment:
The warning message is not proper. I'd suggest something like `"Failed to integrate image: %s", ex`.
In this case, immediately replace the integrator, and set it with the first image:
```python
self._img_intor = img.ImageIntegrator(...)
self._img_intor.append(data)
```
No need to have a `._integration_step`